### PR TITLE
Update Newtonsoft.Json package references to 12.0.2

### PR DESCRIPTION
**Pull request by HenrikNergaard-INS**

### CODE REVIEW

**Comment by HenrikNergaard-INS** [comment]
@manuc66 
It looks like your sonar token is no longer valid.
```SonarScanner.MSBuild.exe : The format of the analysis property sonar.login= is invalid
At line:2 char:5
+     SonarScanner.MSBuild.exe begin /k:"manuc66:JsonSubtypes" /d:"sona ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (The format of t...gin= is invalid:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
The format of the analysis property sonar.github.oauth= is invalid
```

**Comment by manuc66** [comment]
@HenrikNergaard-INS in fact it can not be decrypted from a build coming from a Pull request of another appveyor user because of the encrypted value here : https://github.com/manuc66/JsonSubTypes/blob/05c43b0d505eb34d5dcd1b42571808776aab9dd9/appveyor.yml#L7

I don't know how to fix that without exposing the token publicly which is something I don't want (I you know how to fix that, you're welcome ;-)). 

So don't bother with this error...

File: JsonSubTypes/JsonSubTypes.csproj [hunk 1]
```diff
@@ -23,13 +23,13 @@
...
-    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
-  </ItemGroup>
   <ItemGroup>
     <None Include="..\LICENSE">
       <Pack>True</Pack>
       <PackagePath></PackagePath>
     </None>
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
```
Comment (manuc66) [inline JsonSubTypes/JsonSubTypes.csproj]: What about using floating version ? 

https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#floating-versions

Comment (HenrikNergaard-INS) [reply JsonSubTypes/JsonSubTypes.csproj]: Floating versions should only be used for patch versions, maybe minor versions, but not major ones. From the documentation:
```The Nearest Wins rule can result in a downgrade of the package version, thus potentially breaking other dependencies in the graph. Hence this rule is applied with a warning to alert the user.```

Newtonsoft adds both bugfixes and new features in patch versions so I do not think it is a good idea to wildcard patch versions for this dependency.
https://github.com/JamesNK/Newtonsoft.Json/releases

**Pull request closed without merging**